Command/ExplainAdminCommand.php

```diff
@@ -89,9 +89,18 @@ public function execute(InputInterface $input, OutputInterface $output)
             $output->writeln(sprintf('  - % -25s  % -15s % -15s', $name, $fieldDescription->getType(), $fieldDescription->getTemplate()));
         }
 
+        $metadata = false;
+
         if ($this->getContainer()->has('validator.validator_factory')) {
-            $metadata = $this->getContainer()->get('validator.validator_factory')->getMetadataFor($admin->getClass());
-        } else {
+            $factory = $this->getContainer()->get('validator.validator_factory');
+
+            if (method_exists($factory, 'getMetadataFor')) {
+                $metadata = $factory->getMetadataFor($admin->getClass());
+            }
+        }
+
+        // TODO: remove method check in next major release
```

Comment:
Please replace `TODO` with `NEXT_MAJOR` (we will `grep` for that before releasing)